### PR TITLE
(8) Nouvelle page recherche pro

### PR DESCRIPTION
**Pull request by tcheymol**
[Ticket](https://trello.com/c/mPvZ86cn/3738-8-nouvelle-page-recherche-pro)

### CODE REVIEW

File: tests/v2/Controller/ProControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        array $body = [],
+    ): void {
+        $this->assertRoute($url, $expectedStatusCode, $userMail, $expectedRedirect, $method);
+    }
+
+    public function provideTestRoute(): ?\Generator
+    {
+        yield 'Should redirect to login when not authenticated' => [self::URL, 302, null, '/login'];
+        yield 'Should return 200 status code when authenticated as member' => [self::URL, 200, MemberFixture::MEMBER_MAIL_WITH_RELAYS];
+        yield 'Should return 403 status code when authenticated as member without permissions' => [self::URL, 403, BeneficiaryFixture::BENEFICIARY_MAIL];
```
Comment (AntoineAresu) [inline tests/v2/Controller/ProControllerTest.php]: Wrong user mail

File: templates/v2/pro/search.html.twig [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{% extends 'v2/pro_standard_layout.html.twig' %}
+{% block metaTitle %}{{ 'membre.creationMembre.title'|trans }}{% endblock %}
+
+{% block title %}
+    <span class="text-blue">{{ 'membre.creationMembre.title'|trans }}</span>
```
Comment (AntoineAresu) [inline templates/v2/pro/search.html.twig:5]: We can add new translations

File: templates/v2/pro/_search_results_card.html.twig [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+                {% set user = pro.user %}
+                <li class="list-group-item d-flex justify-content-between align-items-center text-primary">
+                    {{ user.fullName }}
+                    <a class="btn btn-sm btn-blue text-white">{{ 'add'|trans }}</a>
+                </li>
+            {% endfor %}
+        </ul>
+        {{ knp_pagination_render(pros) }}
+    {% endif %}
+{% endblock %}
```
Comment (AntoineAresu) [inline templates/v2/pro/_search_results_card.html.twig:20]: newline

File: templates/v2/pro/create_index.html.twig [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+{% block delimiter %}{% endblock %}
+
+{% block content %}
+    <div class="row">
+        <div class="col-12 col-md-6 mt-2">
+            {{ include('v2/pro/_search_card.html.twig', {form}) }}
+        </div>
+        <div class="col-12 col-md-6 mt-2">
+            {{ include('v2/common/_card.html.twig', {
+                title: 'membre.ajoutMembre.nePossedePasCompte'|trans,
```
Comment (NilsRec) [inline templates/v2/pro/create_index.html.twig:17]: You can use a newer translation

File: tests/v2/Controller/ProControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    ): void {
+        $this->assertRoute($url, $expectedStatusCode, $userMail, $expectedRedirect, $method);
+    }
+
+    public function provideTestRoute(): ?\Generator
+    {
+        yield 'Should redirect to login when not authenticated' => [self::HOME_URL, 302, null, '/login'];
+        yield 'Should return 200 status code when authenticated as member' => [self::HOME_URL, 200, MemberFixture::MEMBER_MAIL_WITH_RELAYS];
+        yield 'Should return 403 status code when authenticated as member without permissions' => [self::HOME_URL, 403, BeneficiaryFixture::BENEFICIARY_MAIL];
+        yield 'Should return 403 status code when authenticated as beneficiary' => [self::HOME_URL, 403, MemberFixture::MEMBER_MAIL];
```
Comment (NilsRec) [inline tests/v2/Controller/ProControllerTest.php:37]: Emails are inverted

**Review by AntoineAresu — APPROVED** [review]

**Review by NilsRec — APPROVED** [review]

**Pull request merged**